### PR TITLE
Global fatal error consumer

### PR DESCRIPTION
**Pull request by cellog**
# Description

In the process of fixing #1093 this PR adds a consumer for the global error context. The next step will be changing withConfig to use this, and concurrently creating one that can be used in the paywall.

Currently this is not used in any components, so no stories yet

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

PS: [Read how to write the perfect pull request](https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/)

### CODE REVIEW

File: unlock-app/src/components/interface/GlobalErrorConsumer.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ */
+export const makeConsumer = (
+  handlers = {},
+  { Consumer } = GlobalErrorContext
+) => {
+  function GlobalErrorConsumer({ children }) {
+    return (
+      <Consumer>
+        {({ error, errorMetadata }) => {
+          if (!error) return <>{children}</>
```
Comment (julien51) [inline unlock-app/src/components/interface/GlobalErrorConsumer.js]: Asking this for y own knownledge by why the need to wrap into `<>`?

File: unlock-app/src/__tests__/components/interface/GlobalErrorConsumer.test.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+const fatalErrors = Object.keys(errors)
+  .reduce(
+    (coll, error) => (error.match(/^FATAL_/) ? [...coll, error] : coll),
+    []
+  )
+  .sort()
+
+describe('global error consumer', () => {
+  it('should have a default handler for every possible fatal error', () => {
+    expect.assertions(1)
```
Comment (julien51) [inline unlock-app/src/__tests__/components/interface/GlobalErrorConsumer.test.js]: 👍

File: unlock-app/src/components/interface/GlobalErrorConsumer.js (hunk 1 above)
Comment (cellog) [reply unlock-app/src/components/interface/GlobalErrorConsumer.js]: if children is multiple components, React needs it to be wrapped in a `React.Fragment` which `<>` is short-hand for. It's a bit Perl, isn't it?

**Comment by cellog** [comment]
So, I have just tried to wire up withConfig, and it's proving to be extremely difficult. I think this PR is the wrong approach. I'm going to close for now, and if I can figure it out tomorrow I'll re-open merge, otherwise I'll use this approach. Turns out the error handlers are very difficult to extend.

**Pull request closed without merging**